src/app/core/header/timepiece/timepiece.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'sh-timepiece',
+    styleUrls: ['./timepiece.component.scss'],
+    templateUrl: './timepiece.component.html'
+})
+
+export class TimepieceComponent implements OnInit, OnDestroy {
+
+    private currentTime: Date;
+    private timer: Observable<Object>;
+    private timeTracker: Subscription;
+
+    ngOnInit() {
+        this.createTimer(0, 1000);
+        this.subscribeToTimer();
+    }
+
+    ngOnDestroy() {
+        this.unsubscribeFromTimer();
+    }
+
+    createTimer(dueTime: number, period: number) {
+        this.timer = Observable.timer(dueTime, period);
+    }
+
+    subscribeToTimer() {
+        this.timeTracker = this.timer.subscribe(() => {
```

Comment:
Do we really need subscribeToTimer and unsubscribeFromTimer methods? It seems that this approach complicates readability of code.